roles/keystone/tasks/main.yml

```diff
@@ -18,35 +18,70 @@
   file: dest=/etc/keystone state=directory
 
 - name: keystone log dir
-  file: dest=/var/log/keystone state=directory mode=0755 owner=keystone
-        group=keystone
-
-- name: install keystone service
-  upstart_service: name=keystone user=keystone cmd=/usr/local/bin/keystone-all
-                   config_dirs=/etc/keystone
+  file: dest=/var/log/keystone state=directory mode=0755 owner=www-data
+        group=www-data
 
 - name: configure keystone
   template: src={{ item }} dest=/etc/keystone/ mode=0644
   with_fileglob: ../templates/etc/keystone/*
   notify:
-    - restart keystone services
+    - reload apache
+
+- name: keystone apache vhost ubuntu 12.04
+  template: src=etc/apache2/sites-available/keystone.conf
+            dest=/etc/apache2/sites-available/keystone
+  when: ansible_distribution_version == "12.04"
+  notify:
+    - reload apache
+
+- name: keystone apache vhost ubuntu
+  template: src=etc/apache2/sites-available/keystone.conf
+            dest=/etc/apache2/sites-available/keystone.conf
+  when: ansible_distribution_version != "12.04"
+  notify:
+    - reload apache
+
+- name: keystone cgi path
+  file: dest=/var/www/cgi-bin/keystone state=directory mode=0755
+        owner=root group=keystone
 
-- name: stop keystone service before db sync
-  service: name=keystone state=stopped
+- name: write out keystone cgi file
+  template: dest=/var/www/cgi-bin/keystone/{{ item }}
+            src=var/www/cgi-bin/keystone/keystone.py
```

Comment:
Should we install this via install from /opt/stack/keystone/httpd/keystone.py rather than copying into Ursula?